app/models/program.rb

```diff
@@ -87,6 +91,17 @@ def languages_list
     self.languages.split(',').map {|l| ISO_639.find(l).english_name} if self.languages.present?
   end
 
+  ##
+  # Checks if there is any event in the program that starts in the given date
+  #
+  # ====Returns
+  # * +True+ -> If there is any event for the given date
+  # * +False+ -> If there is not any event for the given date
+  def any_event_for_this_date?(date)
+    parsed_date = DateTime.parse("#{date} 00:00").utc
+    events.where(start_time: parsed_date..(parsed_date + 1)).any?
```

Comment:
But we  are always saving them as UTC without taking into account timezone:
https://github.com/openSUSE/osem/blob/master/public/javascripts/schedule/schedule.js#L53

I think that this is the correct way to do it, as saving everything related to the schedule in UTC we can forget about time zone changes. The only thing for what we have to take it into account is for opening the schedule in the current day and time, as it may be different to UTC.

If we are always going to save event in UTC maybe we should validate that the events timezone is UTC.